### ISSUE TITLE
Implement JOSS reviewer comments regarding documentation

### ISSUE THREAD

**Issue by fxjung** [PhysicsOfMobility/ridepy#252]
Helpful suggestions by @abhishektiwari:

- [x] Please consider improving the [Github repo](https://github.com/PhysicsOfMobility/ridepy) Readme by adding installation instructions and contribution guidelines. I know these are already covered by [documentation portal](https://ridepy.org/) in great details but given Github repo is landing page for this software paper I highly recommend to update Readme.
- [ ] Please consider improving the A statement of need and Summary covered by [Github repo](https://github.com/PhysicsOfMobility/ridepy) Readme and [documentation portal](https://ridepy.org/) landing page, and align them with paper. I feel they all read differently and only after reading the software paper or the [conceptual overview](https://ridepy.org/overview.html) one can understand the problem statement and target audience.
 - [x] As mentioned by paper ridepy's modular design makes customization easy, while the included modules allow for a quick start, documentation will benefit with examples of how to extend the ridepy modules with a real-world use case. API/code documentation for supported dispatching algorithms partially covers this but additional details will help users.
- [x] Installation instructions worked but can be little bit clear by calling out pre-installation of C++ build environment and the Boost libraries. For instance, on Debian run sudo apt update && sudo apt -y install build-essential libboost-all-dev before pip install ridepy.
